core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -331,52 +365,54 @@ protected void preStart() {
-        }
-    }
+            pollJob = new Callable<HttpToolResponse>() {
+                public HttpToolResponse call() throws Exception {
+                    if (log.isTraceEnabled()) log.trace("http polling for {} sensors at {}", entity, pollInfo);
 
-    // TODO Should we really trustAll for https? Make configurable?
-    private HttpClient createHttpClient(HttpPollIdentifier pollIdentifier) {
-        URI uri = pollIdentifier.uriProvider.get();
-        HttpClientBuilder builder = HttpTool.httpClientBuilder()
-                .trustAll()
-                .laxRedirect(true);
-        if (uri != null) builder.uri(uri);
-        if (uri != null) builder.credential(pollIdentifier.credentials);
-        if (pollIdentifier.connectionTimeout != null) {
-            builder.connectionTimeout(pollIdentifier.connectionTimeout);
-        }
-        if (pollIdentifier.socketTimeout != null) {
-            builder.socketTimeout(pollIdentifier.socketTimeout);
+                    BasicAuth creds = null;
+                    if (pollInfo.credentials.isPresent()) {
+                        creds =  org.apache.brooklyn.util.http.executor.Credentials.basic(
+                                pollInfo.credentials.get().getUserPrincipal().getName(),
+                                pollInfo.credentials.get().getPassword());
+                    }
+
+                    HttpResponse response =  pollInfo.httpExecutor.execute(new HttpRequest.Builder()
+                            .headers(pollInfo.headers)
+                            .uri(pollInfo.uriProvider.get())
+                            .credentials(creds)
+                            .method(pollInfo.method)
+                            .body(pollInfo.body)
+                            .config(HttpConfig.builder()
+                                    .trustSelfSigned(true)
+                                    .trustAll(true)
+                                    .laxRedirect(true)
+                                    .build())
+                            .build());
+                    return createHttpToolRespose(response);
+                }};
+                getPoller().scheduleAtFixedRate(pollJob, new DelegatingPollHandler(handlers), minPeriod);
         }
-        return builder.build();
     }
 
     @SuppressWarnings("unchecked")
     protected Poller<HttpToolResponse> getPoller() {
-        return (Poller<HttpToolResponse>) super.getPoller();
+        return  (Poller<HttpToolResponse>) super.getPoller();
+    }
+
+    @SuppressWarnings("unchecked")
+    private HttpToolResponse createHttpToolRespose(HttpResponse response) throws IOException {
+        int responseCode = response.code();
+        if (responseCode == 400) { // Unprocessable Entity - https://stackoverflow.com/questions/6123425/rest-response-code-for-invalid-data
+            throw new IOException(" Unprocessable Entity: " + response.reasonPhrase());
+        }
```

Comment:
Response code has to be interpreted so it is backward compatible with `HttpToolResponse` which depends on `org.apache.http.HttpResponse`. 
`HttpToolResponse` checks for response code `0` in `getResponseCode()`.

You are correct about code `400` in general. HTTP code `422` will be used instead. I've put a comment from the RFC why it is used.